Providers/StatsProviderConfig.jsx

```diff
@@ -15,15 +17,41 @@ function checkBorders(a, b) {
 class StatsProvider extends Component {
   constructor(props) {
     super(props);
-    this.state = { stats: null };
-    this.loadFromStorage();
+    this.state = { stats: null, date: null };
+  }
+
+  async componentDidMount() {
+    const storedResult = await this.loadFromStorage();
+    try {
+      const resp = await axios.get("http://worldtimeapi.org/api/ip");
+
+      const date = new Date(resp.data.datetime);
+      const storedDate = new Date(storedResult);
+
+      if (this.state.date) {
+        if ((date - storedDate) / 1000 > 86400) {
+          AsyncStorage.setItem("statsDate", JSON.stringify(date));
+          this.setStats(
+            reductionStats.map(a => (a * (date - storedDate)) / 5000)
+          );
+        }
+      } else {
+        this.setState({ date: resp.data.datetime });
+        AsyncStorage.setItem("statsDate", JSON.stringify(date));
+      }
+    } catch (err) {
+      console.log(err);
+    }
   }
 
   loadFromStorage = async () => {
     const stats = await this.getStoredStats();
+    const storedDate = await this.getStoredDate();
     this.setState({
-      stats: stats || defaultStats
+      stats: stats || defaultStats,
+      date: storedDate
```

Comment:
just name storedDate as date, and you can just type `, date`. Not always a good idea, but a handy tip sometimes.

Reply:
or just dont assign to a var beforehand :)